docs/source/community/pull_requests.rst

```diff
@@ -0,0 +1,393 @@
+   - Entirely new documentation sections
+
+#. Patches
+
+   - Largely modifies existing code
+   - Bug fixes, typos, documentation tweaks
+   - Performance optimizations spanning a relatively small amount of code
+
+In general the PR considerations are going to be highly tied to what the PR
+wants to accomplish, the following subsections group considerations by use cases
+and assume archetypal PRs. In other words, not all points are going to be
+relevant to every PR. If they're not relevant for a specific PR they should be
+ignored.
+
+General Considerations
+======================
+
+#. Making a PR should be easy.
+
+   - If it's too hard to make a PR people won't do it.
+   - Setting the bar too high discourages new contributors.
+
+#. It should be possible to open the PR early.
+
+   - Channel for discussion between reviewers and the PR author.
+   - Gets work on radar of reviewers and other developers.
+   - Time to hash out what is and is not in scope before time is wasted
+
+#. The PR's lifetime should be short.
+
+   - Shoot for merging in less than two weeks.
+
+     - Long-lived PRs are typically overlooked in ongoing design
+     - Easy for new changes to break correctness/performance of long-lived PR
+     - Others can't leverage your feature if it's a PR )
+     - PR author needs to keep PR up to date, which can be a lot of time if the
+       PR is long-lived
+     - Hard to remember subtleties over life of long-lived PRs (*e.g.*, why
+       something was/wasn't done)
+     - The longer the PR lives, the more code it tends to contain, and the harder
+       to review
+     - Long-lived PRs tend to increase the amount of technical debt by seemingly
+       justifying the "let's just merge this" mentality
+
+   - Merge function by function if necessary
+   - See :ref:`using_issues_to_track_progress` for tracking progress.
+
+#. Code submitted as PRs should adhere to the organization's standards.
+
+   - Important for continuity
+   - Makes code reviews easier
+   - Current standards:
+
+      - :ref:`cxx_conventions`.
+      - :ref:`python-coding-conventions`.
+      - :ref:`rest_conventions`.
+      - :ref:`doxygen_conventions`.
+
+#. PRs should contain code of high-caliber.
+
+   - NWChemEx strives to be an exemplar package
+   - Easier to maintain good code
+   - High-quality is especially important for senior developers as new
+     developers look to your code for examples
+   - There is a time and place for "just get something working", but should
+     ideally be avoided
+
+API Breaking Changes
+====================
+
+#. Breaking API should be a last resort.
+
+   - Maintaining stable APIs leads to users and developers instilling trust
+     in us.
+   - Need to document what was tried to avoid the break.
+
+#. Need a plan to avoid breaking the APIs again.
+
+   - Determine breakage points
+   - Update design documentation relying on old API.
+   - Carefully plan design of new API to avoid another break
+   - Test new API
```

Comment:
```suggestion
   - Determine breakage points.
   - Update design documentation relying on old API.
   - Carefully plan design of new API to avoid another break.
   - Test new API.
```